redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -187,110 +189,199 @@ inline void
+                    {
+                        if (serviceName.empty())
+                        {
+                            serviceName = service;
+                        }
+                        if (maximumPath.empty())
+                        {
+                            maximumPath = objectPath;
+                        }
+                    }
+                }
+                BMCWEB_LOG_DEBUG << "serviceName: " << serviceName;
+                BMCWEB_LOG_DEBUG << "averagePath: " << averagePath;
+                BMCWEB_LOG_DEBUG << "maximumPath: " << maximumPath;
+            }
+
+            if (objectPath2.empty())
+            {
+                BMCWEB_LOG_DEBUG << "Get power supply date/average/maximum "
+                                    "input power values";
+                getAverageMaximumValues(aResp, serviceName, averagePath,
+                                        maximumPath);
+            }
+            else
+            {
+                getServicePathValues(aResp, objectPath2, "", serviceName,
+                                     averagePath, maximumPath);
+            }
+        };
+
+    crow::connections::systemBus->async_method_call(
+        getServiceAndPathHandler, "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject", objectPath,
+        interfaces);
+}
+
 /**
- * @brief Get power supply average, maximum and date values given chassis and
- * power supply IDs.
+ * @brief Get power supply average, maximum and date values given chassis
+ * and power supply IDs.
  *
  * @param[in] aResp - Shared pointer for asynchronous calls.
- * @param[in] chassisID - Chassis to which the values are associated.
- * @param[in] powerSupplyID - Power supply to which the values are associated.
+ * @param[in] inputHistoryItem - Array of object paths for input history.
  *
  * @return None.
  */
 inline void getValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
-                      const std::string& chassisID,
-                      const std::string& powerSupplyID)
+                      const std::vector<std::string>& inputHistoryItem)
 {
-    BMCWEB_LOG_DEBUG
-        << "Get power supply date/average/maximum input power values";
+    BMCWEB_LOG_DEBUG << "ENTER: getValues(...)";
+    for (const auto& item : inputHistoryItem)
+    {
+        BMCWEB_LOG_DEBUG << " inputHistoryItem: " << item;
+    }
+
     // Setup InputPowerHistoryItems values array.
     // It will have 0 to many date/timestamp, average, and maximum entries.
     aResp->res
         .jsonValue["Oem"]["IBM"]["InputPowerHistoryItems"]["@odata.type"] =
         "#OemPowerSupplyMetric.InputPowerHistoryItems";
 
-    const std::array<const char*, 2> interfaces = {
-        "org.open_power.Sensor.Aggregation.History.Average",
-        "org.open_power.Sensor.Aggregation.History.Maximum"};
+    std::string serviceName;
```

Comment:
This is kind of confusing, since we are sending parameters with no values to that function.  In reading the code, I understand how it works since the function is called multiple times.  I guess given time limitations maybe it is fine.

Reply:
I did have a case where I was doing this bad with the asynch, and I ended up with this blank. I was not sure if that was really bad or not. I suppose it could end up that way if you boot up or power on and read this less than 30 seconds later.